graphql/query_resolver.go

```diff
@@ -338,14 +338,23 @@ func (r *queryResolver) Hosts(ctx context.Context, hostID *string, distroID *str
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting hosts: %s", err.Error()))
 	}
 
-	apiHosts := []*restModel.APIHost{}
+	usr := mustHaveUser(ctx)
+	forbiddenHosts := []string{}
+	for _, h := range hosts {
+		if !userHasHostPermission(usr, h.Distro.Id, evergreen.HostsView.Value) {
+			forbiddenHosts = append(forbiddenHosts, h.Id)
+		}
+	}
+	if len(forbiddenHosts) > 0 {
+		return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access one or more hosts.", usr.Username()))
```

Comment:
since we don't really do anything with the `forbiddenHosts`, I wonder if it makes sense to do this check in the same loop where we're building the APIHosts below (and just return immediately if you encounter no permissions)

Reply:
You can either do that or include a formatted list of all the forbidden hosts in the error message [here](https://github.com/evergreen-ci/evergreen/pull/8337/files#diff-18582cc382c2280a0446b8fa448fd4ca997753d6ebfe6042a981701b312164c6R349)

Reply:
I originally considered returning a list of hosts that are inaccessible but I think that is insecure since the query parameters match the hostIds in the error message.